src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,138 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(last, value, null);
+        if (size == 0) {
+            first = newNode;
+        } else {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        isIndexValid(index);
+        Node<T> oldNode = getNode(index);
```

Comment:
This is code repeating. You always check index before getting node by index. What you should do is move the check logic INSIDE the get method.
![image](https://github.com/mate-academy/jv-linked-list/assets/20326858/2b7ac1b3-ee0a-4230-83c1-db7dfa88c555)